postinstall.pl

```diff
@@ -0,0 +1,23 @@
+use strict;
+use warnings;
+our ($module_name, %gconfig);
+
+do 'virtualmin-htpasswd-lib.pl';
+
+sub module_install
+{
+# Enable virtualmin-htpasswd module once
+if (&virtual_server::plugin_defined("virtualmin-nginx", "start_nginx")) {
+    if ($virtual_server::config{'plugins'} !~ /$module_name/ &&
+        $virtual_server::config{'plugins_postinstall_enabled'} !~ /$module_name/) {
+        &virtual_server::lock_file($virtual_server::module_config_file);
+        $virtual_server::config{'plugins'} .= " $module_name";
+        $virtual_server::config{'plugins'} =~ s/^\s+|\s+$//g;
+        $virtual_server::config{'plugins_postinstall_enabled'} .= " $module_name";
+        $virtual_server::config{'plugins_postinstall_enabled'} =~ s/^\s+|\s+$//g;
```

Comment:
Why not use split into an array and then join instead of these regexp operations on `plugins` and `plugins_postinstall_enabled` ?

Reply:
It could also be done using `split` and `join`. But I think working with strings is more simple.

Reply:
Yeah but it's not really a string, it's list of values. Using split / join would remove the need for line 17 to cleanup trailing spaces.

Reply:
Better to write it like : 

```
my @p = split(/\s+/, $virtual_server::config{'plugins'});
my @ppe = split(/\s+/, $virtual_server::config{'plugins_postinstall_enabled'});
if (&indexof('virtualmin-nginx', @virtual_server::plugins) >= 0 &&
    &indexof($module_name, @p) < 0 &&
    &indexof($module_name, @ppe) < 0)) {
        push(@p, $module_name);
        push(@ppe, $module_name);
        $virtual_server::config{'plugins'} = join(" ", @p);
        $virtual_server::config{'plugins_postinstall_enabled'} = join(" ", @ppe);
        &virtual_server::save_module_config();
        }
```

Reply:
Yes it's a string, but when Virtualmin interprets it we treat it as a space-separated list. So all manipulation should be done in the same way to avoid the issue of leading or trailing spaces. It's not avoid computational code since this code is barely ever run - it's about good coding practices.

Reply:
> It's not avoid computational code since this code is barely ever run - it's about good coding practices.

I agree with the importance of consistency in how we read and store data! However, if we consistently expect space-separated values to be read into an array and stored as a string of space-separated values, perhaps we should consider implementing a dedicated API for that purpose.

The fix you proposed earlier is here https://github.com/virtualmin/virtualmin-htpasswd/pull/6/commits/d9ff2568e82070c56c3b45a03689112d87a7da87. Thank you!

Reply:
Thanks! And yes it would be nice if lists in the module config were parsed into array refs, but that's a more complex problem ...